invenio_stats/tasks.py

```diff
@@ -38,3 +43,142 @@ def process_events(event_types):
+            yield dict(_index='{0}-{1}-{2}'.
+                       format(self.time_frame,
+                              self.event,
+                              datetime.date.today().strftime('%Y-%m')),
+                       _type='{0}_bookmark'.format(self.event),
+                       _source=bookmark)
+
+        bulk(self.client,
+             _success_date(),
+             stats_only=True)
+
+    def pending_dates(self):
+        """Get all dates that haven't been processed."""
+        start_date = self.get_bookmark()
+        end_date = datetime.date.today()
+        import ipdb
+        ipdb.set_trace()
+        if start_date >= end_date:
+            for n in range((start_date - end_date).days + 1):
+                yield start_date + datetime.timedelta(n)
+        else:
+            for n in range((end_date - start_date).days + 1):
+                yield start_date - datetime.timedelta(n)
+
+    def aggr_iter(self, date):
+        """Aggregate and return dictionary to be indexed in ES."""
+        aggregation_data = {}
+        result = self.query.execute()
+
+        import ipdb
+        ipdb.set_trace()
+
+        for aggregation in result.aggregations:
+            for hit in result.aggregations[aggregation]['buckets']:
+                aggregation_data['bucket'] = hit['key']
+                aggregation_data['count'] = hit['doc_count']
+                print('aggregation_data:', aggregation_data)
+
+                yield dict(_id='{0}_{1}'.
+                           format(hit['key'],
+                                  date),
+                           _index='{0}-{1}-{2}'.
+                           format(self.time_frame,
+                                  self.event,
+                                  date.strftime('%Y-%m')),
+                           _type='{0}_aggregation'.
+                           format(self.event),
+                           _source=aggregation_data)
+
+    def get_indices_with_alias(self, alias_name):
+        """Get all indices under an alias."""
+        result = []
+        all_indices = current_search_client.indices.get_aliases()
+        for index in all_indices:
+            if alias_name in all_indices[index]['aliases']:
+                result.append(index)
+        result.sort()
+        return result[0]
+
+    def run(self):
+        """Calculate statistics aggregations."""
+        for date in self.pending_dates():
+            print("Aggregating date:", date)
+            index_name = 'events-stats_{0}-{1}'.format(
+                self.event, date.isoformat().replace('-', '.'))
+            if not Index(index_name, using=current_search_client).exists():
+                continue
+
+            self.query = Search(using=current_search_client,
+                                index=index_name)
+            self.query.aggs.bucket('by_{}'.format(self.event),
+                                   'terms', field='bucket',
+                                   size=999999)
```

Comment:
As said Friday, we need to be careful here. We should be able to handle the case where there are thousands of files which have been downloaded, and I don't think that Elasticsearch will scale well with that many buckets.